pkg/standalone-utils/test/GaugeActions.test.ts

```diff
@@ -0,0 +1,659 @@
+            recipient: tokenRecipient,
+            amount: withdrawAmountOrRef,
+          }),
+        ]);
+
+        receipt = await tx.wait();
+      });
+
+      // Short-circuit when no tokens are withdrawn.
+      expectedAmount.gt(0) &&
+        it('pulls gauge tokens from sender if necessary', async () => {
+          // This transfer is skipped if token sender is relayer, but addresses are undefined outside this scope.
+          if (tokenSenderAddress != relayer.address) {
+            expectTransferEvent(
+              receipt,
+              { from: tokenSenderAddress, to: relayer.address, value: expectedAmount },
+              gauge.address
+            );
+          }
+        });
+
+      // Short-circuit when no tokens are withdrawn.
+      expectedAmount.gt(0) &&
+        it('emits BPT transfer event from gauge to relayer', async () => {
+          expectTransferEvent(
+            receipt,
+            { from: gauge.address, to: relayer.address, value: expectedAmount },
+            lpToken.address
+          );
+        });
+
+      it('emits withdraw event', async () => {
+        expectEvent.inIndirectReceipt(receipt, gauge.interface, 'Withdraw', {
+          provider: relayer.address,
+          value: expectedAmount,
+        });
+      });
+
+      it('burns gauge tokens', async () => {
+        expect(await gauge.balanceOf(tokenSenderAddress)).to.be.almostEqual(totalLpTokens.sub(expectedAmount));
+      });
+
+      it('emits transfer event for burned gauge tokens', async () => {
+        expectTransferEvent(receipt, { from: relayer.address, to: ZERO_ADDRESS, value: expectedAmount }, gauge.address);
+      });
+
+      it('emits BPT transfer event from relayer to recipient if necessary', async () => {
+        // This transfer is skipped if token recipient is relayer, but addresses are undefined outside this scope.
+        if (tokenRecipientAddress != relayer.address) {
+          expectTransferEvent(
+            receipt,
+            { from: relayer.address, to: tokenRecipientAddress, value: expectedAmount },
+            lpToken.address
+          );
+        }
+      });
+
+      it('transfers BPT tokens to recipient', async () => {
+        if (tokenRecipientAddress == tokenSenderAddress) {
+          expect(await lpToken.balanceOf(tokenSenderAddress)).to.be.eq(expectedAmount);
+        } else {
+          expect(await lpToken.balanceOf(tokenSenderAddress)).to.be.eq(0);
+        }
+        expect(await lpToken.balanceOf(tokenRecipientAddress)).to.be.eq(expectedAmount);
+        expect(await lpToken.balanceOf(gauge.address)).to.be.almostEqual(totalLpTokens.sub(expectedAmount));
+      });
+    }
+  });
+
+  describe('gaugeMint', () => {
+    sharedBeforeEach('grant mint approval to sender via relayer', async () => {
+      const { v, r, s, deadline } = await BalancerMinterAuthorization.signSetMinterApproval(
+        balMinter,
+        relayer.address,
+        true,
+        userSender
+      );
```

Comment:
It'd be even better if you had this in the same multicall as the call to mint :)

Reply:
(which is the expected usage)

Reply:
done :)